### PR TITLE
Bump morphio

### PR DESCRIPTION
**Pull request by wizmer**

### CODE REVIEW

**Comment by wizmer** [comment]
Hi, 
I would not mind a little help to debug this.
I have tried reproducing the compilation error by doing:
```
cd morphio # my local morphio repository
spack setup morphio@2.3.4~mpi
# edit spconfig.py to add -DBUILD_BINDINGS:BOOL=ON
mkdir build
cd build
/gpfs/bbp.cscs.ch/apps/hpc/jenkins/deploy/external-libraries/2018-12-19/linux-rhel7-x86_64/gcc-6.4.0/python-3.6.5-ukuow6gd2f/bin/python ../spconfig.py .. # The shebang is too long to lauch the script as an executable
make -j
```
But the software builds perfectly. I don't know what I am doing wrong.

**Comment by pramodk** [comment]
@wizmer : note that PR is building fine but when built modules are tested then it's failing:

image

In [hpc.module-testing]( 
https://bbpcode.epfl.ch/ci/blue/organizations/jenkins/hpc.module-testing/detail/hpc.module-testing/1416/pipeline/) I see:

image

which doesn't explain why `config/modules.sh` file is not generated properly in PR. Let me re-trigger.

File: var/spack/repos/builtin/packages/py-morphio/package.py [hunk 1]
```diff
@@ -10,9 +10,11 @@ class PyMorphio(PythonPackage):
     """Python library for reading / writing morphology files"""
 
     homepage = "https://github.com/BlueBrain/MorphIO/"
+    url      = "https://github.com/BlueBrain/MorphIO.git"
     git      = "https://github.com/BlueBrain/MorphIO.git"
 
-    version('develop', branch='master', submodules=True, clean=False)
+    version('develop', git=url, submodules=True, clean=False)
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/py-morphio/package.py]: Why this change?

**Review by matz-e — COMMENTED** [review]
Do we need this PR? It seems that your other one already takes care of this.

File: var/spack/repos/builtin/packages/py-morphio/package.py (hunk 1 above)
Comment (wizmer) [reply var/spack/repos/builtin/packages/py-morphio/package.py]: Since the `morphio` package was working, I took it from there to see if that would solve the issue. In the meantime, I found out the main difference between `morphio` and `py-morphio` is that the later has BUILD_BINDINGS=ON.
TLDR: I could revert that, I guess

**Comment by wizmer** [comment]
Yes, I think it's better to fix this first and then I'll proceed with the remaining battles to build NeuroC.
